ibm/resource_ibm_pi_instance.go

```diff
@@ -268,41 +211,55 @@ func resourceIBMPIInstance() *schema.Resource {
 				Description: "PI instance image id",
 			},
 			helpers.PIInstanceProcessors: {
-				Type:        schema.TypeFloat,
-				Required:    true,
-				Description: "Processors count",
+				Type:          schema.TypeFloat,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Processors count",
 			},
 			helpers.PIInstanceName: {
 				Type:        schema.TypeString,
 				Required:    true,
 				Description: "PI Instance name",
 			},
 			helpers.PIInstanceProcType: {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validateAllowedStringValue([]string{"dedicated", "shared", "capped"}),
-				Description:  "Instance processor type",
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  validateAllowedStringValue([]string{"dedicated", "shared", "capped"}),
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Instance processor type",
 			},
 			helpers.PIInstanceSSHKeyName: {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "SSH key name",
 			},
 			helpers.PIInstanceMemory: {
-				Type:        schema.TypeFloat,
-				Required:    true,
-				Description: "Memory size",
+				Type:          schema.TypeFloat,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Memory size",
+			},
+			PISAPInstanceProfileID: {
```

Comment:
sap_profile_id , and all other 4 (memory, etc) are all made optional.. Isn't either of one required? or will this resource work even if user doesn't provide any of these 5 arguments?

Reply:
It is not required for SAP instances hence the change. I have made it optional and checking it dynamically in the `createPVMInstance` function where it is needed.